core/core-funcs.el

```diff
@@ -297,19 +297,13 @@ buffer."
   "Switch back and forth between current and last buffer in the
 current window."
   (interactive)
-  (let ((current-buffer (window-buffer window))
-        (buffer-predicate
-         (frame-parameter (window-frame window) 'buffer-predicate)))
+  (let ((current-buffer (window-buffer window)))
     ;; switch to first buffer previously shown in this window that matches
     ;; frame-parameter `buffer-predicate'
     (switch-to-buffer
-     (or (cl-find-if (lambda (buffer)
-                       (and (not (eq buffer current-buffer))
-                            (or (null buffer-predicate)
-                                (funcall buffer-predicate buffer))))
-                     (mapcar #'car (window-prev-buffers window)))
-         ;; `other-buffer' honors `buffer-predicate' so no need to filter
-         (other-buffer current-buffer t)))))
```

Comment:
`other-buffer` is used when there isn't any suitable buffer in `window-prev-buffers`. However, it's ok to remove it because it wasn't needed in the first place. If `cl-find-if` returns `nil`, then `switch-to-buffer` will default to calling `other-buffer` (because it got `nil` as its BUFFER argument). It's worth a comment in the code, though.